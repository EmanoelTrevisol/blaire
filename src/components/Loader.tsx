import React from 'react';
import { View, StyleSheet, ActivityIndicator } from 'react-native';

export enum Sizes {
  LARGE = 'large',
  SMALL = 'small',
}

export interface IProps {
  size?: Sizes;
  color?: string;
}

const Loader = ({ size = Sizes.LARGE, color = '#000' }: IProps) => {
  return (
    <View style={stl.container}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};

const stl = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Loader;
