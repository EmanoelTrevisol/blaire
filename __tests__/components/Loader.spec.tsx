import React from 'react';
import { render } from '@testing-library/react-native';
import { ActivityIndicator } from 'react-native';
import Loader, { IProps, Sizes } from '@components/Loader';

describe('Loader.tsx', () => {
  const renderComp = (props?: IProps) => {
    return render(<Loader {...props} />);
  };

  test('matches snapshot', () => {
    const { toJSON } = renderComp(undefined);

    expect(toJSON()).toMatchSnapshot();
  });

  test('passes default props to ActivityIndicator', () => {
    const { UNSAFE_getByType } = renderComp(undefined);

    const loader = UNSAFE_getByType(ActivityIndicator);

    expect(loader).toBeDefined();
    expect(loader.props.size).toBe('large');
    expect(loader.props.color).toBe('#000');
  });

  test('passes props correctly ActivityIndicator', () => {
    const size = Sizes.SMALL;
    const color = '#ff0000';

    const { UNSAFE_getByType } = renderComp({ size, color });

    const loader = UNSAFE_getByType(ActivityIndicator);

    expect(loader).toBeDefined();
    expect(loader.props.size).toBe(size);
    expect(loader.props.color).toBe(color);
  });
});
