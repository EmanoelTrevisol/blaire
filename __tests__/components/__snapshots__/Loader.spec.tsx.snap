// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.tsx matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#000"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
